curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/understanding-the-differences-between-the-freecodecamp-and-browser-console.english.md

```diff
@@ -7,31 +7,28 @@ forumTopicId: 301193
-First, the browser console.  When you load and run an ordinary JavaScript file in your browser the <code>console.log()</code> statements will print exactly what you tell them to print to the browser console the exact number of times you requested. In your in-browser text editor the process is slightly different and can be confusing at first.
-Values passed to <code>console.log()</code> in the text editor block run each set of tests as well as one more time for any function calls that you have in your code.
-This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times depending on the number of tests and the values being passed to those tests.
-If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code>.
+You may have noticed that some freeCodeCamp JavaScript challenges include their own console. This console behaves a little differently than the browser console you used in the last challenge.
+The following challenge is meant to highlight the main difference between the freeCodeCamp console and your browser console.
+When you run ordinary JavaScript, the browsers console will display your <code>console.log()</code> statements the exact number of times you requested.
+The freeCodeCamp console will print your <code>console.log()</code> statements for each test of that challenge, as well as one more time for any function calls that you have in your code.
+This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times.
+If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code> and check the browsers console.
 </section>
 
 ## Instructions
 <section id='instructions'>
-Use <code>console.log()</code> to print the variables in the code where indicated.
-
+First, use <code>console.clear()</code> to clear the browser console. After that, use <code>console.log</code> to log the <code>output</code> variable.
 </section>
 
 ## Tests
 <section id='tests'>
 
 ```yml
 tests:
-  - text: Use <code>console.log()</code> to print the <code>outputTwo</code> variable.  In your Browser Console this should print out the value of the variable two times.
-    testString: assert(code.match(/console\.log\(outputTwo\)/g));
-  - text: Use <code>console.log()</code> to print the <code>outputOne</code> variable.
-    testString: assert(code.match(/console\.log\(outputOne\)/g));
-  - text: Use <code>console.clear()</code> to modify your output so that <code>outputOne</code> variable only outputs once.
+  - text: Use <code>console.clear()</code> to clear the browser console.
     testString: assert(code.match(/^(\s*console.clear\(\);?\s*)$/gm));
```

Comment:
To allow for whites space can we use a helper function?  Also, if we are making the semi-colon optional, might as well get rid of it. I left them in, but do we really need the `^` and `$` here?
```suggestion
    testString: const noSpaces = code.replace(/\s/g, ''); assert(noSpaces.match(/^console.clear\(\)$/gm));
```

Reply:
I didn't really take a look at the tests, just changed what was there to check for the new variable name. No, we probably don't need the `^` or `?` or the `m`.

If you want to change your suggestion, I can commit them here.

Reply:
I have edited both of my original suggestions.